plasTeX/ConfigManager.py

```diff
@@ -0,0 +1,181 @@
+        group.add_argument(*self.options, dest=self.name, type=self.valueType(), help=self.description + formatDefault(self.value))
+
+    def valueType(self) -> Type:
+        return type(self.value)
+
+    def updateFromDict(self, data: Dict["str", Any]):
+        value = data.get(self.name)
+        if value is not None:
+            self.value = value
+
+    def setFromString(self, string: str):
+        self.value = self.valueType()(string)
+
+    @property
+    def value(self) -> T:
+        return self._value
+
+    @value.setter
+    def value(self, value: T):
+        if type(value) != self.valueType():
+            raise TypeError("Encountered {} when expecting {}".format(type(value).__name__, self.valueType().__name__))
+
+        self._value = value
+
+class BooleanOption(Option[bool]):
+    def registerArgparse(self, group: ArgumentGroup):
+        enables = [x for x in self.options if x[0] != "!"]
+        disables = [x[1:] for x in self.options if x[0] == "!"]
+
+        group.add_argument(*enables, dest=self.name, help=self.description + formatDefault(self.value), action='store_true', default=None)
+        if len(disables) > 0:
+            group.add_argument(*disables, dest=self.name, help=self.description + formatDefault(self.value), action='store_false', default=None)
+
+class MultiStringOption(Option[List[str]]):
+    def registerArgparse(self, group: ArgumentGroup):
+        group.add_argument(*self.options, dest=self.name, type=str, nargs="*", help=self.description, action="append")
+
+    def setFromString(self, string: str):
+        self.value.extend(shlex.split(string))
+
+    def updateFromDict(self, data: Dict["str", Any]):
+        value = data.get(self.name)
+        if value is not None:
+            for entry in value:
+                self._value.extend(entry)
+
+class DictOption(ABC, Option[Dict[str, T]]):
+    @classmethod
+    @abstractmethod
+    def entryFromString(cls, entry: str) -> T:
+        """This convers a string into T"""
+
+    def set(self, key: str, value: str):
+        self.value[key] = self.entryFromString(value)
+
+    def setFromString(self, string: str):
+        raise ValueError("Calling setFromString on a DictOption. The individual items should be inserted with set.")
+
+    def updateFromDict(self, data: Dict["str", Any]):
+        entries = data.get(self.name)
+        if entries is not None:
+            for key, val in entries:
+                self.set(key, val)
+
+class ConfigSection():
+    def __init__(self, name: str, parent: 'ConfigManager'):
+        self.name = name
+        self.parent = parent
+        self.data = {} # type: Dict[str, Option]
+
+    def __getitem__(self, key: str) -> Any:
+        value = self.data[key].value
+        wrapper = InterpolationWrapper(self.parent)
+        if isinstance(value, str):
+            return value % wrapper
+        elif isinstance(value, list) and len(value) > 0 and isinstance(value[0], str):
```

Comment:
```suggestion
        elif isinstance(value, list) and value and isinstance(value[0], str):
```
Also, do you really want a list here, or `Sequence` would be enough?

Reply:
Sequence would be enough.